packages/dids/src/methods/did-ion.ts

```diff
@@ -300,26 +379,80 @@ export class DidIon extends DidMethod {
-    // verification methods.
-    const keySet = await DidIon.generateKeys({
-      keyManager,
-      verificationMethods: options.verificationMethods ?? []
+    // If no verification methods were specified, generate a default Ed25519 verification method.
+    const defaultVerificationMethod: DidCreateVerificationMethod<TKms> = {
+      algorithm : 'Ed25519' as any,
+      purposes  : ['authentication', 'assertionMethod', 'capabilityDelegation', 'capabilityInvocation']
+    };
+
+    const verificationMethodsToAdd: DidIonVerificationMethod[] = [];
+
+    // Generate random key material for additional verification methods, if any.
+    for (const vm of options.verificationMethods ?? [defaultVerificationMethod]) {
+      // Generate a random key for the verification method.
+      const keyUri = await keyManager.generateKey({ algorithm: vm.algorithm });
+      const publicKey = await keyManager.getPublicKey({ keyUri });
+
+      // Add the verification method to the DID document.
+      verificationMethodsToAdd.push({
+        id           : vm.id,
+        publicKeyJwk : publicKey,
+        purposes     : vm.purposes ?? ['authentication', 'assertionMethod', 'capabilityDelegation', 'capabilityInvocation']
+      });
+    }
+
+    // Generate a random key for the ION Recovery Key. Sidetree requires secp256k1 recovery keys.
+    const recoveryKeyUri = await keyManager.generateKey({ algorithm: DidIonRegisteredKeyType.secp256k1 });
+    const recoveryKey = await keyManager.getPublicKey({ keyUri: recoveryKeyUri });
+
+    // Generate a random key for the ION Update Key. Sidetree requires secp256k1 update keys.
+    const updateKeyUri = await keyManager.generateKey({ algorithm: DidIonRegisteredKeyType.secp256k1 });
+    const updateKey = await keyManager.getPublicKey({ keyUri: updateKeyUri });
+
+    // Compute the Long Form DID URI from the keys and services, if any.
+    const longFormDidUri = await DidIonUtils.computeLongFormDidUri({
+      recoveryKey,
+      updateKey,
+      services            : options.services ?? [],
+      verificationMethods : verificationMethodsToAdd
     });
 
-    // Create the DID object from the generated key material, including DID document, metadata,
-    // signer convenience function, and URI.
-    const did = await DidIon.fromPublicKeys({ keyManager, options, ...keySet });
+    // Expand the DID URI string to a DID document.
+    const { didDocument, didResolutionMetadata } = await DidIon.resolve(longFormDidUri, { gatewayUri: options.gatewayUri });
+    if (didDocument === null) {
+      throw new Error(`Unable to resolve DID during creation: ${didResolutionMetadata?.error}`);
+    }
+
+    // Create the BearerDid object, including the "Short Form" of the DID URI, the ION update and
+    // recovery keys, and specifying that the DID has not yet been published.
+    const did = new BearerDid({
+      uri      : longFormDidUri,
+      document : didDocument,
+      metadata : {
+        published   : false,
```

Comment:
is there something in the options to flip this to true?

Reply:
No -- that is the state prior to publishing.  If `options.publish` is true or undefined (since publishing is done by default) then the did.metadata will be updated on the outcome of the attempt to publish.  If publishing fails, then `published` will remain `false`.  If publishing succeeds, `published` will be set to `true`.